subsys/net/lib/mqtt_helper/mqtt_helper.c

```diff
@@ -383,7 +384,24 @@ static int broker_init(struct sockaddr_storage *broker,
 	addr = result;
 
 	while (addr != NULL) {
-		if (addr->ai_addrlen == sizeof(struct sockaddr_in)) {
+		/* Check for IPv6 first and use if available. */
```

Comment:
Actually it doesn't matter if you check IPv4 or IPv6 first here, because Zephyr `getaddrinfo()` will always give IPv4 address first and IPv6 address second. If you want to prefer IPv6 you should look in the last `addr` entry first.